neurodamus/utils/pyutils.py

```diff
@@ -5,11 +5,37 @@
 import subprocess  # noqa: S404
 import weakref
 from bisect import bisect_left
-from enum import EnumMeta
+from enum import EnumMeta, IntEnum
 
 import numpy as np
 
 
+class StrEnumBase(IntEnum):
+    __mapping__: list[tuple[str, int]] = []
+    # default for when there is value. Leaving None throws an error
+    __default__ = None
+    # default when the string is not found in the mapping
+    __invalid__ = None
+
+    @classmethod
+    def from_string(cls, s: str):
+        if not s:
+            return cls(cls.__default__)
+        mapping = dict(cls.__mapping__)
+        return cls(mapping.get(s.lower(), cls.__invalid__ if cls.__invalid__ is not None else s))
+
+    def to_string(self) -> str:
```

Comment:
I hadn't noticed this before, but these methods are strangely named.
`__str__`'s job is `to_string`; and it looks to me that `__str__` is actually `__repr__` in this case.
Doesn't need to be fixed in this PR, just surprised by it.

Reply:
Let's not change how StrEnumBase works in this PR.

However, the fact that `StrEnumBase` inherits from `IntEnum` is already weird, IMO; but again, separate issue.

Reply:
I would have the "nice" names everywhere, except at the single point where the permutation is set for neuron, where the is a dict that maps the "nice" version to the required number.

That keeps the code easier to read, imo, and it's only at the interface where the conversion to the neuron specified requirement happens.

Reply:
but it is like that. the `to_string` is to get the string for neuron compatibility. Do you want a better name? Or you think the neuron one is the nice one? Should I override the str to return something like repr?